lab_1/main.py

```diff
@@ -86,10 +167,44 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
+    if (not isinstance(unknown_profile, dict)
+            or not isinstance(profile_to_compare, dict)
+            or not isinstance(top_n, int)):
+        return None
+    # use function get_top_n_words to get common and sorted_common
+    top_n_words_unknown = get_top_n_words(unknown_profile["freq"], top_n)
+    top_n_words_compare = get_top_n_words(profile_to_compare["freq"], top_n)
+    common = []
+    for word in top_n_words_compare:
+        if word in top_n_words_unknown:
+            common.append(word)
+    # common = list(set(top_n_words_compare) & set(top_n_words_unknown)) I can't use this here:(
```

Comment:
why do you think you can't?

Reply:
I could have done this if I wasn't required to return common words without sorting in the final dictionary named "report", but by the condition of the task I need to do this. So I had to avoid sets here, because they are unordered, and the elements contained in the set will be added to the list ("common") by random sequence. Here we need the order.